### PR TITLE
Tensornet implementation

### PR DESCRIPTION
**Pull request by wiederm**
## Description
Implement tensornet using the parts that @MarshallYan has already built and implemented.

## Todos
  - [x] Setting up representation and embedding
  - [x] Implementing message creation
  - [x] Implement message passing

## Status
- [x] Ready to go

### CODE REVIEW

File: devtools/conda-envs/test_env.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ dependencies:
   - torchvision
   - openff-units
   - torchmetrics>=1.4
-  - pint
+  - pint=0.23
```
Comment (wiederm) [inline devtools/conda-envs/test_env.yaml:20]: is this pin necessary?

File: modelforge/potential/utils.py [hunk 2]
```diff
@@ -1177,6 +1177,81 @@ def nondimensionalize_distances(self, distances: torch.Tensor) -> torch.Tensor:
...
+    ):
+        super().__init__(
+            number_of_radial_basis_functions,
+            max_distance,
+            min_distance,
+            alpha,
+            dtype,
+            trainable_centers_and_scale_factors,
+        )
+
```
Comment (wiederm) [inline modelforge/potential/utils.py:1198]: You can remove this. As @ianmkenney mentioned, if you don't overwrite the __init__ method in the child class the __init__ of the super class is used

File: devtools/conda-envs/test_env.yaml (hunk 1 above)
Comment (MarshallYan) [reply devtools/conda-envs/test_env.yaml:21]: This is necessary on my local setup.

File: modelforge/potential/utils.py (hunk 2 above)
Comment (MarshallYan) [reply modelforge/potential/utils.py]: Solved.

File: devtools/conda-envs/test_env.yaml (hunk 1 above)
Comment (ArnNag) [reply devtools/conda-envs/test_env.yaml:21]: I was also having issues with openff-units and pint, not sure if that's related. I had to pin openff-units>=0.2.0

Comment (wiederm) [reply devtools/conda-envs/test_env.yaml:21]: we don't see this in CI, that's strange

Comment (wiederm) [reply devtools/conda-envs/test_env.yaml:21]: @chrisiacovella , have you seen this?

**Pull request merged**